framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -143,106 +260,170 @@ class MeshGenerator : public MooseObject, public MeshMetaDataInterface
    * caller doesn't specify a value for \p dim, then the value in the \p Mesh input file block will
    * be used
    */
-  std::unique_ptr<DistributedMesh> buildDistributedMesh(unsigned int dim = libMesh::invalid_uint);
+  [[nodiscard]] std::unique_ptr<DistributedMesh>
+  buildDistributedMesh(unsigned int dim = libMesh::invalid_uint);
 
   /**
    * Construct a "subgenerator", a different MeshGenerator subclass
    * that will be added to the same MooseApp on the fly.
    *
```

Comment:
```suggestion
   *
   * Notes:
   * - The new generator will generate() before its parent.
   * - Sub-generators must be added in the order that they are executed.
   * - MeshGeneratorName parameters of the parent that are inputs of subs must be declared as such
```

Reply:
feel free to add more I dont think we gave enough advice on this new capability in the code